### PR TITLE
Feature/ele 588 planning item assignment list improvements

### PR DESCRIPTION
**Pull request by dannelundqvist**
Added useLink hook to simplify opening views.
Added generic prompt dialog to simplify using prompt modals.
Use useYValue() instead of useYObserver() to simplify

### CODE REVIEW

File: src/components/DataItem/AssignmentType.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const AssignmentType = ({ path, editable = false }: {
     return <></>
   }
 
-  const value = state.map((s) => s.value).join('/')
+  const value = state.map((s) => s.value).sort().join('/')
```
Comment (gularsson) [inline src/components/DataItem/AssignmentType.tsx:20]: :+1:

File: src/hooks/useLink.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import type { KeyboardEvent, MouseEvent } from 'react'
+import { useNavigation, useView } from '@/hooks'
+import { handleLink } from '@/components/Link/lib/handleLink'
+
+export const useLink = (viewName: 'Editor' | 'PlanningOverview' | 'Planning') => {
```
Comment (gularsson) [inline src/hooks/useLink.tsx:5]:
```suggestion
import { View } from 'what-ever-path'
export const useLink = (viewName: View) => {
```
Would this work? Then we don't need to change the types when/if we add views?

File: src/views/Planning/components/AssignmentRow.tsx [hunk 3]
```diff
@@ -1,288 +1,224 @@
...
+      <AssignmentRowContent
+        index={index}
+        setSelectedAssignment={setSelectedAssignment}
+      />
+    </div>
   )
 }
 
-const AssignmentRowContent = ({ index, setSelectedAssignment, setShowCreateDialog, showCreateDialog = false }: {
+const AssignmentRowContent = ({ index, setSelectedAssignment /*, setShowCreateDialog, showCreateDialog = false */ }: {
```
Comment (gularsson) [inline src/views/Planning/components/AssignmentRow.tsx:43]: Remove comment?

File: src/views/Planning/components/AssignmentRow.tsx [hunk 4]
```diff
@@ -1,288 +1,224 @@
...
-  setShowCreateDialog?: React.Dispatch<React.SetStateAction<boolean>>
-  showCreateDialog?: boolean
 }): JSX.Element => {
-  const { get } = useYObserver('meta', `core/assignment[${index}]`)
-  const { get: getAssignmentDescription } = useYObserver('meta', `core/assignment[${index}].meta.core/description[0].data`)
-  const { get: getAssignmentPublishTime } = useYObserver('meta', `core/assignment[${index}].data`)
+  const { provider } = useCollaboration()
+  const openArticle = useLink('Editor')
+  const base = `meta.core/assignment[${index}]`
+  // const [id] = useYValue<string>(`${assPath}.id`)
```
Comment (gularsson) [inline src/views/Planning/components/AssignmentRow.tsx:50]: Remove?

**Review by gularsson — CHANGES_REQUESTED** [review]
useLink and Prompt looks great, some minor things.

File: src/hooks/useLink.tsx (hunk 2 above)
Comment (dannelundqvist) [reply src/hooks/useLink.tsx:5]: Yes, that is a good idea! Otherwise this will be a headache in the future. I'll add this as a separate issue though,

**Review by gularsson — APPROVED** [review]
👍

**Pull request merged**